### ISSUE TITLE
no-fallthrough flagging last case + reporting wrong line number

### ISSUE THREAD

**Issue by nzakas** [eslint/eslint#416]
There are two issues with this rule.

First, it incorrectly flags `case` statements when they are the last thing in the `switch`, such as:

``` js
switch(foo) {
    case "bar":
       console.log(0);
}
```

Second, it incorrectly specifies the node for the `switch` rather than the node for the `case`, so the line number reported isn't useful.

cc @mduvall
